File: Algos/src/uniqueOrder.js
var uniqueInOrder=function(iterable){
    // proměná result je prázdné pole
    let result = [];
    // procházíme vstupní array
    for (let i = 0; i < iterable.length; i++) {
        // pokud je prvek na indexu i různý od prvku na indexu i + 1
        if (iterable[i] !== iterable[i + 1]) {
            // pushujeme prvek na indexu i do výsledného pole
            result.push(iterable[i]);
        }
    }
    return result;
}

console.log(uniqueInOrder('AAAABBBCCDAABBB'));
console.log(uniqueInOrder('ABBCcAD'));
console.log(uniqueInOrder([1,2,2,3,3]));



// Implement the function unique_in_order which takes as argument a sequence and returns a list of items without any
// elements with the same value next to each other and preserving the original order of elements.

//     For example:
// iniqueInOrder('AAAABBBCCDAABBB') == ['A', 'B', 'C', 'D', 'A', 'B']
// uniqueInOrder('ABBCcAD')         == ['A', 'B', 'C', 'c', 'A', 'D']
// uniqueInOrder([1,2,2,3,3])       == [1,2,3]

File: Algos/src/UniqueOrder.java
public class UniqueOrder {

    public static void main(String[] args) {
        System.out.println(uniqueInOrder("AAAABBBCCDAABBB"));
        System.out.println(uniqueInOrder("ABBCcAD"));
        System.out.println(uniqueInOrder("12233"));
        System.out.println(uniqueInOrder("AB"));
        System.out.println(uniqueInOrder("A"));
        System.out.println(uniqueInOrder(""));
    }

    public static String uniqueInOrder(String iterable) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < iterable.length(); i++) {
            if (i == 0 || iterable.charAt(i) != iterable.charAt(i - 1)) {
                sb.append(iterable.charAt(i));
            }
        }
        return sb.toString();
    }
}
